libqfieldsync/offline_converter.py

```diff
@@ -149,21 +151,30 @@ def _convert(self, project: QgsProject) -> None:
         on_original_project_read = self._on_original_project_read_wrapper(
             xml_elements_to_preserve
         )
-        project.readProject.connect(on_original_project_read)
 
+        temporary_project = QgsProject()
+        temporary_project_filename = ""
         if self.export_type == ExportType.Cable:
             # the `backup_filename` is copied right after packaging is requested. It has all the unsaved
             # project settings, which means they will be available in the packaged project too.
-            project.read(self.backup_filename)
+            temporary_project_filename = self.backup_filename
         elif self.export_type == ExportType.Cloud:
             # if you save the project without QGIS GUI, the project no longer has `theMapCanvas` canvas
             # so we should use the original project file that already has `theMapCanvas`. There is no
             # gain using the `backup_filename`, since there is no user to modify the project.
-            project.read(project.fileName())
+            temporary_project_filename = project.fileName()
         else:
             raise NotImplementedError(f"Unknown package type: {self.export_type}")
 
-        project.readProject.disconnect(on_original_project_read)
+        read_flags = QgsProject.ReadFlags()
+        read_flags |= QgsProject.FlagDontResolveLayers
+        read_flags |= QgsProject.FlagDontLoadLayouts
+        if Qgis.versionInt() >= 32600:
+            read_flags |= QgsProject.FlagDontLoad3DViews
+
+        temporary_project.readProject.connect(on_original_project_read)
```

Comment:
The signal handler sets "theMapCanvas" object. Is the map canvas configured correctly when we store the XML in a temporary filename?

Reply:
@suricactus , yep, this isn't a problem.